examples/fungible-token/src/lib.rs

```diff
@@ -1,553 +1,117 @@
-    pub fn get_balance(&self, owner_id: AccountId) -> U128 {
-        self.get_account(&owner_id).balance.into()
-    }
-
-    /// Returns current allowance of `escrow_account_id` for the account of `owner_id`.
-    ///
-    /// NOTE: Other contracts should not rely on this information, because by the moment a contract
-    /// receives this information, the allowance may already be changed by the owner.
-    /// So this method should only be used on the front-end to see the current allowance.
-    pub fn get_allowance(&self, owner_id: AccountId, escrow_account_id: AccountId) -> U128 {
-        assert!(
-            env::is_valid_account_id(escrow_account_id.as_bytes()),
-            "Escrow account ID is invalid"
-        );
-        self.get_account(&owner_id).get_allowance(&escrow_account_id).into()
+        let mut this = Self { token: FungibleToken::new(b"a") };
+        this.token.internal_register_account(owner_id.as_ref());
+        this.token.internal_deposit(owner_id.as_ref(), total_supply.into());
+        this
     }
 }
 
-impl FungibleToken {
-    /// Helper method to get the account details for `owner_id`.
-    fn get_account(&self, owner_id: &AccountId) -> Account {
-        assert!(env::is_valid_account_id(owner_id.as_bytes()), "Owner's account ID is invalid");
-        let account_hash = env::sha256(owner_id.as_bytes());
-        self.accounts.get(&account_hash).unwrap_or_else(|| Account::new(account_hash))
-    }
-
-    /// Helper method to set the account details for `owner_id` to the state.
-    fn set_account(&mut self, owner_id: &AccountId, account: &Account) {
-        let account_hash = env::sha256(owner_id.as_bytes());
-        if account.balance > 0 || account.num_allowances > 0 {
-            self.accounts.insert(&account_hash, &account);
-        } else {
-            self.accounts.remove(&account_hash);
-        }
-    }
-
-    fn refund_storage(&self, initial_storage: StorageUsage) {
-        let current_storage = env::storage_usage();
-        let attached_deposit = env::attached_deposit();
-        let refund_amount = if current_storage > initial_storage {
-            let required_deposit =
-                Balance::from(current_storage - initial_storage) * STORAGE_PRICE_PER_BYTE;
-            assert!(
-                required_deposit <= attached_deposit,
-                "The required attached deposit is {}, but the given attached deposit is is {}",
-                required_deposit,
-                attached_deposit,
-            );
-            attached_deposit - required_deposit
-        } else {
-            attached_deposit
-                + Balance::from(initial_storage - current_storage) * STORAGE_PRICE_PER_BYTE
-        };
-        if refund_amount > 0 {
-            env::log(format!("Refunding {} tokens for storage", refund_amount).as_bytes());
-            Promise::new(env::predecessor_account_id()).transfer(refund_amount);
-        }
-    }
-}
+near_contract_standards::impl_fungible_token!(
```

Comment:
I think this should split into 3 impl still:
for example if contract wants to programmatically set the metadata, would need to implement all these functions separately.
Also in contract that is token + something else, impl of StorageManager would probably be more sophisticated.

Reply:
I'm curious if there is a way to have a generic proxy for traits with composing. I feel like I saw a crate like this.
Something like `#[compose(token)] impl StorageManager for Contract {}`

Reply:
I've decided to remove Metadata macro completely, since it doesn't help much, but instead hides named arguments from the user, so it's easier to mess up.

There were 3 independent macro before combined `impl_fungible_token` macro. Now only `_core` and `_storage` left. Updated the example